config/config.go

```diff
@@ -1014,6 +1015,12 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if (c.Action == RelabelReplace || c.Action == RelabelHashMod) && c.TargetLabel == "" {
 		return fmt.Errorf("relabel configuration for %s action requires 'target_label' value", c.Action)
 	}
+	if c.Action == RelabelReplace && !relabelTarget.Match([]byte(c.TargetLabel)) {
+		return fmt.Errorf("%q is invalid 'target_label' for %s action", c.TargetLabel, c.Action)
+	}
+	if c.Action == RelabelHashMod && !model.LabelNameRE.Match([]byte(c.TargetLabel)) {
```

Comment:
Hmm.. I think I understand a bit better now, I'm unsure whether it is possible for us to validate at this point when we allow interpolation, as we don't know if for example the interpolation causes the label to start with `[a-zA-Z]`.